ddtrace/tracer/payload.go

```diff
@@ -121,6 +123,27 @@ func (p *unsafePayload) push(t []*Span) (stats payloadStats, err error) {
 	return p.stats(), nil
 }
 
+func (p *unsafePayload) setTracerTags(t []*Span) {
+	// set on first chunk
+	if atomic.LoadUint32(&p.count) != 0 {
+		return
+	}
+	pTags := processtags.GlobalTags().String()
+	if pTags == "" {
+		return
+	}
+	for _, s := range t {
+		// todo: can any span be nil?
+		if s == nil {
+			continue
+		}
+		s.mu.Lock()
+		s.meta[keyProcessTags] = pTags
+		s.mu.Unlock()
```

Comment:
I would rather have this snippet inside a helper method defined in span.go as part of Span type. This is not documented yet, but we want to consolidate all the lock usage closer to where the type defined in a single file to make it easier to manage.

```suggestion
		s.setProcessTags(pTags)
```